templates/al2023/provisioners/install-nvidia-driver.sh

```diff
@@ -7,22 +7,58 @@ set -o errexit
 if [ "$ENABLE_ACCELERATOR" != "nvidia" ]; then
   exit 0
 fi
+
+# utility function for pulling rpms from an S3 bucket
+function rpm_install() {
+  local RPMS=($@)
+  echo "pulling and installing rpms: (${RPMS[@]}) from s3 bucket: (${BINARY_BUCKET_NAME}) in region: (${BINARY_BUCKET_REGION})"
+  for RPM in ${RPMS[@]}; do
+    aws s3 cp --region ${BINARY_BUCKET_REGION} s3://${BINARY_BUCKET_NAME}/rpms/${RPM} ${WORKING_DIR}/${RPM}
+    sudo yum localinstall -y ${WORKING_DIR}/${RPM} 
+  done
+}
+
+#Detect Isolated partitions
+function is-isolated-partition() {
+  PARTITION=$(imds /latest/meta-data/services/partition)
+  NON_ISOLATED_PARTITIONS=("aws" "aws-cn" "aws-us-gov")
+  for NON_ISOLATED_PARTITION in "${NON_ISOLATED_PARTITIONS[@]}"; do
+    if [ "${NON_ISOLATED_PARTITION}" = "${PARTITION}" ]; then
+      return 1
+    fi
+  done
+  return 0
+}
+
 echo "Installing NVIDIA ${NVIDIA_DRIVER_MAJOR_VERSION} drivers..."
 
 ################################################################################
 ### Add repository #############################################################
 ################################################################################
 # Determine the domain based on the region
-if [[ $AWS_REGION == cn-* ]]; then
-  DOMAIN="nvidia.cn"
+if is-isolated-partition; then
+  echo "[amzn2023-nvidia]
+  name=Amazon Linux 2023 Nvidia repository
+  mirrorlist=https://al2023-repos-$awsregion-de612dc2.s3.$awsregion.$awsdomain/nvidia/mirrors/$releasever/$basearch/mirror.list
```

Comment:
need to present these from being substituted using single quotes?

Reply:
yea i think using single instead of double quotes to encase this blob should suffice, the neuron excerpt doesn't have any `$VAR` items in it that might get replaced with a shell variable like this `mirrorlist` value